connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignorTest.java

```diff
@@ -1375,24 +1171,6 @@ private void assertCompleteAllocation() {
         });
     }
 
-    private void verifyCoordinatorInteractions() {
-        verify(coordinator, times(rebalanceNum)).configSnapshot();
-        verify(coordinator, times(rebalanceNum)).leaderState(any());
-        verify(coordinator, times(2 * rebalanceNum)).generationId();
-        verify(coordinator, times(rebalanceNum)).memberId();
-        verify(coordinator, times(rebalanceNum)).lastCompletedGenerationId();
-    }
```

Comment:
After this refactor, we can't test the coordinator interaction. Do we have other tests for it?

Reply:
Not at the moment, no. The existing tests don't give us that much anyways:

- It doesn't matter at all how many times we request the current config snapshot, generation ID, member ID, or last completed generation ID; these methods all have trivial implementations
- Although it does matter that we invoke `Coordinator::leaderState` during the rebalance, the existing tests only verify that we invoke it an expected number of times, but without any assertions about what the actual state given to the coordinator is

Reply:
I agree all the verification is not much helpful, except the `leaderState`. Could we add some tests to verify the `Coordinator.leaderState` after `performTaskAssignment`? I think Mock is also fine, just want to make sure it is well tested. Thank you.

Reply:
Ack, done. The new test does nothing to verify the content of the leader state, but it does verify that `WorkerCoordinator::leaderState` is invoked, which is the same coverage that the current tests provide.

I've also augmented the existing test logic to check the content of the `ClusterAssignment` (specifically, the return value of the `allAssignedConnectors` and `allAssignedTasks` methods for each worker in the cluster) in `applyAssignments`. As long as we use the `ClusterAssignment` object correctly (and it's fairly trivial to verify that we do) by passing its state to the coordinator, then this should ensure that the proper leader state is recorded.